vortex-array/src/array/sparse/mod.rs

```diff
@@ -78,7 +76,7 @@ impl SparseArray {
 
     /// Return indices as a vector of usize with the indices_offset applied.
     pub fn resolved_indices(&self) -> Vec<usize> {
-        flatten_primitive(cast(self.indices(), PType::U64.into()).unwrap().as_ref())
+        flatten_primitive(self.indices())
```

Comment:
You need the cast here, it's unsafe to access typed_data without asserting/enforcing the type of the indices.

Reply:
indices type is asserted in constructor